src/utils/config.ts

```diff
@@ -439,34 +439,108 @@ export function configAutoResolve(
-
-    if (!locals) {
+  return Object.entries(all).reduce((acc, [key, allLocals]) => {
+    if (!allLocals) {
       return acc;
     }
 
-    const isEnabled = Object.keys(locals).find(key => {
-      if (!metrics.includes(key)) {
-        return false;
+    const locals = Object.entries(allLocals).reduce((acc, [key, local]) => {
+      if (metrics && !metrics.includes(key)) {
+        return acc;
       }
-      const local = locals[key];
-      if (local.enabled === false) {
-        return Object.values(local.alarm || {}).find(l => l.enabled === true) !== undefined;
+
+      if (!configIsEnabled(local)) {
+        return acc;
       }
-      return Object.values(local.alarm || {}).find(l => l.enabled !== false) !== undefined;
-    });
 
-    // Add only if at least one value is enabled
-    if (isEnabled) {
-      return { ...acc, [key]: locals };
-    }
+      return {
+        ...acc,
+        [key]: {
+          ...local,
+          alarms: Object.entries(local.alarm || {}).reduce((acc, [topic, alarm]) => {
+            if ((local.enabled === false && alarm.enabled === true) || alarm.enabled !== false) {
+              return {
+                ...acc,
+                [topic]: alarm,
+              };
+            }
+
+            return acc;
+          }, {} as TopicMap<AlarmOptions>),
+        }
+      }
+    }, {} as T);
 
     // Skip adding as none was enabled
-    return acc;
+    if (Object.keys(locals).length === 0) {
+      return acc;
+    }
+
+    return { ...acc, [key]: locals };
   }, {});
 }
+
+/**
+ * Calculate how many resources each metric alarm (example lambda) takes
+ */
+export function calculateResources<T extends ConfigMetricAlarms = ConfigMetricAlarms>(
+  confType: ConfigLocalType,
+  allowedMetrics?: string[],
+  addPerResource = 0,
+): { [key: string]: [number, T] } {
+  const enabled = configGetAllEnabled<T>(confType, allowedMetrics);
+  const metricCount = 1;
+
+  const acc: { [key: string]: [number, T] } = {};
+  for (const [key, metrics] of Object.entries(enabled)) {
+    for (const [metric, local] of Object.entries(metrics)) {
+      const [currentCount, currentLocals] = acc[key] || [0, {}]
+
+      const alarmCount = Object.values(local.alarm || {}).length;
+      acc[key] = [currentCount + alarmCount + metricCount + addPerResource, {
+        ...currentLocals,
+        [metric]: local
+      }];
+    }
+  }
+
+  return acc;
+}
+
+/**
+ * Generate chunked array of the metric alarms
+ */
+export function chunkByStackLimit<T extends ConfigMetricAlarms = ConfigMetricAlarms>(
+  confType: ConfigLocalType,
+  allowedMetrics?: string[],
+  addedPerResource = 0,
+): ConfigLocals<T>[] {
+  // Calculate before chunking
+  const results = calculateResources<T>(confType, allowedMetrics, addedPerResource);
+
+  const acc: ConfigLocals<T>[] = []
+  let currentCount = 0;
+  let index = 0;
+  for (const [key, [count, metrics]] of Object.entries(results)) {
+    if ((currentCount + count) > 500) {
```

Comment:
If I'm not mistaken looking at these conditions, `CDKMetadata` is not taken into consideration despite using a resource slot in the stack resources (unless version reporting is explicitly disabled). This could result in incorrect resource count calculation.

For example - 100 lambdas with 5 alerts would generate 500 resources + `CDKMedatada`, so 501 in total, but the code would sum it as 500 and wouldn't shift the index - which would result in CDK attempting to deploy 501 resources and failing in such attempt.

Reply:
Latest change fixed this by actually finding if the version reporting is enabled from the context. It won't be included in the calculation, but it will decrease the max resource count by one if it is enabled so the `CDKMetada` can fit in to  the stack.